### PR TITLE
Added parameters i and even to limit the number of candidate keys of ECRECOVER

### PR DESCRIPTION
**Pull request by hbs**

### CODE REVIEW

File: warp10/src/main/java/io/warp10/script/functions/ECRECOVER.java [hunk 1]
```diff
@@ -46,6 +46,8 @@
 
   private static final String KEY_R = "r";
   private static final String KEY_S = "s";
+  private static final String KEY_I = "i";
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/script/functions/ECRECOVER.java:49]: Why don't you call that parameter `h`?

File: warp10/src/main/java/io/warp10/script/functions/ECRECOVER.java [hunk 2]
```diff
@@ -169,9 +171,15 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
 
     BigInteger rinv = spec.getCurve().fromBigInteger(r.modInverse(spec.getN())).toBigInteger();
 
+    int minH = 0;
     int maxH = H.intValue();
 
-    if (maxH > MAX_COFACTOR) {
+    if (params.get(KEY_I) instanceof Long) {
+      minH = ((Long) params.get(KEY_I)).intValue();
+      maxH = minH;
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/script/functions/ECRECOVER.java:179]: There must be a difference of 1 between those two for the `for` loop to execute. I'd do `minH = ((Long) params.get(KEY_I)).intValue() - 1;`.

**Review by ftence — CHANGES_REQUESTED** [review]

File: warp10/src/main/java/io/warp10/script/functions/ECRECOVER.java (hunk 1 above)
Comment (hbs) [reply warp10/src/main/java/io/warp10/script/functions/ECRECOVER.java:49]: h is the actual cofactor of the curve, it is a fixed value, i is an index between 0 and the cofactor.

File: warp10/src/main/java/io/warp10/script/functions/ECRECOVER.java (hunk 2 above)
Comment (hbs) [reply warp10/src/main/java/io/warp10/script/functions/ECRECOVER.java:179]: It's better if maxH is set to minH + 1, because the loop starts at minH.

**Review by ftence — APPROVED** [review]

**Pull request merged**